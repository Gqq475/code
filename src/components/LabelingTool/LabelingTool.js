import React from 'react'
import { connect } from 'react-redux'
import classNames from 'classnames'
import classes from './LabelingTool.scss'
import { Button } from '@blueprintjs/core'
import Tabs from 'components/LabelingToolDefectType'
import LabelingToolImagesZone from 'components/LabelingToolImagesZone'
import LabelingToolDefectCodeList from 'components/LabelingToolDefectCodeList'
import LabelingToolCategoryDialog from 'components/LabelingToolCategoryDialog'
import OurToaster from 'components/OurToaster'
import {
  SELECT_DEFECTPICTURE_SOURCE,
  MARK_DEFECT,
  SELECT_DEFECT_MARK,
  INITIALIZE_LABELING_TOOL
} from 'routes/OJS/modules/OJS'

type Props = {
  data: Object,
  getLabelingToolImageList: Function,
  getLabelingToolImageDefect: Function,
  defectPictureSource: String,
  selectDefectPictureSource: Function,
  selectDefect: Function,
  defectList: Array,
  markDefect: Function,
  getDefectTypeList: Function,
  addDefectType: Function,
  token: String,
  // dispatch: Function,
  emptyLabelingTool: Function
};

export class LabelingTool extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
    this.toggleDialog = this.toggleDialog.bind(this)
    this.getImagesList = this.getImagesList.bind(this)
    this.requestJudge = this.requestJudge.bind(this)
    this.state = {
      isOpen: false,
      selected: '',
      pageNo: 1,
      pageSize: 10
    }
  }

  handleClick (e) {
    let { index, id } = e.target.dataset
    index *= 1
    this.setState({
      selected: e.target.dataset.index
    })
    this.props.selectDefectPictureSource(this.props.data.labelingToolImageList[index])
    this.props.getLabelingToolImageDefect(this.state, id)
  }

  toggleDialog () {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  getImagesList () {
    this.props.getLabelingToolImageList()
  }

  componentWillUnmount () {
    this.props.emptyLabelingTool()
  }

  requestJudge () {
    let { id } = this.props.defectPictureSource
    fetch(`${__QCA__}/offline/labeling/image/defect/xmlfile?imageId=${id}`, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': this.props.token
      }
    })
      .then(response => response.json())
      .then(json => {
        if (json.code === 200000) {
          OurToaster.show({message: '判定成功!'})
        } else {
          OurToaster.show({message: '判定失败!'})
        }
      })
  }

  render () {
    return (
      <div className={classes['LabelingTool-container']}>
        <div>
          <Button className={`pt-button pt-intent-primary ${classes.btnFile}`} onClick={this.getImagesList}>
             文件
          </Button>
          <Button className={`pt-button pt-intent-primary ${classes.btn}`} disabled>
             文件夹
          </Button>
        </div>
        <div className={classes.imgList}>
          <ul>
            {
              this.props.data.labelingToolImageList &&
              this.props.data.labelingToolImageList.map((item, index) => {
                let itemClass = classNames({
                  [classes.imgItem]: true,
                  [classes.active]: this.state.selected === index + ''
                })
                return (
                  <li
                    onClick={this.handleClick}
                    className={itemClass}
                    key={`${index}`}
                    data-index={index}
                    data-id={item.id}>
                    {item.name}
                  </li>
                )
              })
            }
          </ul>
        </div>
        <LabelingToolDefectCodeList
          defectList={this.props.defectList}
          selectDefect={this.props.selectDefect}
          />
        <LabelingToolImagesZone
          widthOfOriginalImage={400}
          heightOfOriginalImage={300}
          defectList={this.props.defectList}
          markDefect={this.props.markDefect}
          defectPictureSource={this.props.defectPictureSource}
          selectDefect={this.props.selectDefect}
          />
        <div className={classes.decidsion}>
          <Tabs data={this.props.data} getDefectTypeList={this.props.getDefectTypeList} />
          <div className={classes.btnGroup}>
            <Button className={classes.category} onClick={this.toggleDialog}>类别修改</Button>
            <Button onClick={this.requestJudge}>判定</Button>
            <LabelingToolCategoryDialog isOpen={this.state.isOpen} toggleDialog={this.toggleDialog}
              addDefectType={this.props.addDefectType}
              data={this.props.data} getDefectTypeList={this.props.getDefectTypeList}
            />
          </div>
        </div>
      </div>
    )
  }
}

export default connect(
 (state) => ({
   defectPictureSource: state.ojs.defectPictureSource,
   defectList: state.ojs.defectList,
   currentDefectIndex: state.ojs.currentDefectIndex,
   token: state.auth.token
 }),
 (dispatch) => ({
   selectDefectPictureSource: (sourceObj) => {
     dispatch({
       type: SELECT_DEFECTPICTURE_SOURCE,
       data: { ...sourceObj }
     })
   },
   markDefect: (data) => {
     dispatch({
       type: MARK_DEFECT,
       data
     })
   },
   selectDefect: (rowIndex) => {
     dispatch({
       type: SELECT_DEFECT_MARK,
       data: rowIndex
     })
   },
   emptyLabelingTool: () => {
     dispatch({
       type: INITIALIZE_LABELING_TOOL
     })
   }
 })
)(LabelingTool)
