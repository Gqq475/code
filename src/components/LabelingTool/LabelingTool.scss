.LabelingTool-container {
  position: absolute;
  top: 20px;
  left: 20px;
  right: 20px;
  height: 100%;
  background: #fff;
  ul,li {
    margin: 0px;
    padding: 0px;
  }
  .btn {
    width: 80px;
    margin-bottom: 15px;
  }
  .btnFile {
    margin: 0px 15px 15px 0px;
    width: 80px;
  }
  .imgList{
    width: 60%;
	  float: left;
    overflow: hidden;
    border: 1px solid #0087DC;
    border-radius: 5px;
    box-sizing: border-box;
    padding: 0 0 10px 0;
    height: 215px;
  }
  .imgItem {
    padding: 5px 0px 5px 5px;
    cursor: pointer;
  }
  .active {
    background:#b0e0e6;
    color: #666;
  }
  .decidsion {
    width: 39%;
    float: right;
  }
  .decidsion .btnGroup {
    float: right;
  }
  .decidsion .category {
    margin-right: 10px;
  }
  .decidsion button{
    width: 80px;
  }
}
