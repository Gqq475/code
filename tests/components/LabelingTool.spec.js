import React from 'react'
import { LabelingTool } from 'components/LabelingTool/LabelingTool'

describe('(Component) LabelingTool', () => {
  it('render normally', () => {
    const data = {
      labelingToolImageList: [{
        createTime: 'Apr 27, 2017 11:19:11 AM',
        depth: 10,
        height: 10,
        id: 1,
        name: 'Image 01',
        path: 'ttp://10.120.136.214/doc/qms/sample_defect_image/TPDPD.jpg',
        type: 'Type 1',
        width: 10
      }]
    }
    const wrapper = shallow(
      <LabelingTool data={data} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
