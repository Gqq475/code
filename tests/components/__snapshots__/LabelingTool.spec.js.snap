// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) LabelingTool render normally 1`] = `
<div
  className="LabelingTool-container"
>
  <div>
    <Blueprint.Button
      className="pt-button pt-intent-primary btnFile"
      onClick={[Function]}
    >
      文件
    </Blueprint.Button>
    <Blueprint.Button
      className="pt-button pt-intent-primary btn"
      disabled={true}
    >
      文件夹
    </Blueprint.Button>
  </div>
  <div
    className="imgList"
  >
    <ul>
      <li
        className="imgItem"
        data-id={1}
        data-index={0}
        onClick={[Function]}
      >
        Image 01
      </li>
    </ul>
  </div>
  <LabelingToolDefectCodeList />
  <LabelingToolImagesZone
    heightOfOriginalImage={300}
    widthOfOriginalImage={400}
  />
  <div
    className="decidsion"
  >
    <LabelingToolDefectType
      data={
        Object {
          "labelingToolImageList": Array [
            Object {
              "createTime": "Apr 27, 2017 11:19:11 AM",
              "depth": 10,
              "height": 10,
              "id": 1,
              "name": "Image 01",
              "path": "ttp://10.120.136.214/doc/qms/sample_defect_image/TPDPD.jpg",
              "type": "Type 1",
              "width": 10,
            },
          ],
        }
      }
    />
    <div
      className="btnGroup"
    >
      <Blueprint.Button
        className="category"
        onClick={[Function]}
      >
        类别修改
      </Blueprint.Button>
      <Blueprint.Button
        onClick={[Function]}
      >
        判定
      </Blueprint.Button>
      <Connect(LabelingToolCategoryDialog)
        data={
          Object {
            "labelingToolImageList": Array [
              Object {
                "createTime": "Apr 27, 2017 11:19:11 AM",
                "depth": 10,
                "height": 10,
                "id": 1,
                "name": "Image 01",
                "path": "ttp://10.120.136.214/doc/qms/sample_defect_image/TPDPD.jpg",
                "type": "Type 1",
                "width": 10,
              },
            ],
          }
        }
        isOpen={false}
        toggleDialog={[Function]}
      />
    </div>
  </div>
</div>
`;
